app/src/main/java/org/p2p/wallet/transaction/progresshandler/ClaimProgressHandler.kt

```diff
@@ -0,0 +1,37 @@
+package org.p2p.wallet.transaction.progresshandler
+
+import timber.log.Timber
+import org.p2p.core.glide.GlideManager
+import org.p2p.uikit.utils.getString
+import org.p2p.wallet.R
+import org.p2p.wallet.transaction.model.progressstate.ClaimProgressState
+import org.p2p.wallet.transaction.model.progressstate.TransactionState
+
+class ClaimProgressHandler(
+    glideManager: GlideManager
+) : TransactionProgressHandler(glideManager) {
+
+    companion object {
+        const val QUALIFIER = "ClaimHandler"
+    }
+
+    override fun handleState(state: TransactionState) {
+        super.handleState(state)
+        when (state) {
+            is ClaimProgressState.Success -> setSuccessState()
+            is ClaimProgressState.Error -> setErrorState()
+            else -> {
+                Timber.d("Unsupported State: $state")
```

Comment:
it seems there should be error, because it means someone's passing a state to the inappropriate handler
```suggestion
                Timber.e("Unsupported State for ClaimProgressHandler: $state")
```

Reply:
or this log could be removed if it won't happen at all